src/checkout/checkout-store-error-selector.ts

```diff
@@ -295,6 +306,19 @@ export function createCheckoutStoreErrorSelectorFactory(): CheckoutStoreErrorSel
         }
     );
 
+    const getInitialError = createSelector(
```

Comment:
I see this as a convenience method for getting flash messages. It's nice to have that because it's a bit of a hassle to traverse the `ContextConfig` object in order to retrieve a message. But considering we do have other types of flash messages and they can get set because of a prior action before a redirect, I'm wondering if it's better if we have a method that can return other types of message as well. This is so that we don't end up with other variants of this method to expose other types of messages in the future.

```ts
getFlashMessages(type?: FlashMessageType): FlashMessage[]
```

Reply:
I think flash messages in general are quite random and the name itself "flash message" does not tell much to the consumers. Also, I'm not sure what do we use them for aside for errors (in checkout). That's why I thought about exposing only error flash messages initially, using `getInitialError` selector.

If you think is better to anticipate use cases, I'm happy to have a `getFlashMessages` selector, maybe as part of the `data` selector?
`state.data.getStoreConfig()`
`state.data.getFlashMessages()`

currently is
`state.errors.getInitialError()`

thoughts?

Reply:
Yes, I understand where you're coming from. But we can't hide the fact that they are modelled as flash messages as they are already exposed through `config.context.flashMessages`.

Anyway, if we are going to have this convenience method, yes, I agree we should have it under `data` instead.